__tests__/testRouter.test.js

```diff
@@ -0,0 +1,45 @@
+const fs = require("fs");
+const path = require("path");
+const { handleTestEvent } = require("../testRouter");
+
+const inputDataFile = fs.readFileSync(
+  path.resolve(__dirname, "./data/test_router_input.json")
+);
+const outputDataFile = fs.readFileSync(
+  path.resolve(__dirname, "./data/test_router_output.json")
+);
+
+const inputData = JSON.parse(inputDataFile);
+const outputData = JSON.parse(outputDataFile);
+
+
+describe('Test Router Tests', () => {
+  it('should fail with error on receiving wrong type for events', () => {
+    try {}
+    catch (error) {
+      expect(error.message).toEqual("events array is required in payload");
+    }
+  });
+
+  describe('API Response tests', () => {
+    inputData.forEach((input, index) => {
+      const dest = input.destination.destinationDefinition.name;
+      const ctxMock = {
+        request: {
+          body: {
+            events: [input]
+          }
+        }
+      };
+      it(`Input: ${index} should return proper response`, async () => {
+        try {
+          await handleTestEvent(ctxMock, dest.toLowerCase());
+          expect(ctxMock.body).toEqual(outputData[index]);
+        }
+        catch (err) {
+          // should not enter this block
```

Comment:
Say there is an exception with a case or due to a change, I think the test-case will get passed because we've caught the error already. Can you verify if that is the case ?

Reply:
yes actually. should not be wrapping test function call in try-catch block. will update this part.